module/x/gravity/keeper/keeper.go

```diff
@@ -664,22 +664,20 @@ func (k Keeper) DeleteEthereumSignatures(ctx sdk.Context, otx types.OutgoingTx)
 // This just does keeper state cleanup if a new gravity contract has been deployed
 func (k Keeper) MigrateGravityContract(ctx sdk.Context, newBridgeAddress string, bridgeDeploymentHeight uint64) {
 	// Delete Any Outgoing TXs.
-
 	prefixStoreOtx := prefix.NewStore(ctx.KVStore(k.storeKey), []byte{types.OutgoingTxKey})
 	iterOtx := prefixStoreOtx.ReverseIterator(nil, nil)
 	defer iterOtx.Close()
 	for ; iterOtx.Valid(); iterOtx.Next() {
+		prefixStoreOtx.Delete(iterOtx.Key())
+	}
 
-		var any cdctypes.Any
-		k.cdc.MustUnmarshal(iterOtx.Value(), &any)
-		var otx types.OutgoingTx
-		if err := k.cdc.UnpackAny(&any, &otx); err != nil {
-			panic(err)
-		}
-		// Delete any partial Eth Signatures handging around
-		k.DeleteEthereumSignatures(ctx, otx)
```

Comment:
the eth signature is store with the key
```
<EthereumSignatureKeyPrefix><OtxNonce>
```

this part of the code is just looping over all the otx and deleting the ethereum signatures using the prefixed iterator
```
prefixStoreSig := prefix.NewStore(ctx.KVStore(k.storeKey), append([]byte{types.EthereumSignatureKey}, otx.GetStoreIndex()...))
```